### ISSUE TITLE
Shapefilepy fails to open a shapefile that has zero records.

### ISSUE THREAD

**Issue by jesegal** [GeospatialPython/pyshp#83]
Error when opening a shapefile that has zero records.

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python27\ArcGIS10.3\lib\site-packages\shapefile.py", line 234, in __init__
    self.load(args[0])
  File "C:\Python27\ArcGIS10.3\lib\site-packages\shapefile.py", line 279, in load
    self.__dbfHeader()
  File "C:\Python27\ArcGIS10.3\lib\site-packages\shapefile.py", line 476, in __dbfHeader
    fmt,fmtSize = self.__recordFmt()
  File "C:\Python27\ArcGIS10.3\lib\site-packages\shapefile.py", line 482, in __recordFmt
    self.__dbfHeader()
  File "C:\Python27\ArcGIS10.3\lib\site-packages\shapefile.py", line 460, in __dbfHeader
    fieldDesc = list(unpack("<11sc4xBB14x", dbf.read(32)))
error: unpack requires a string argument of length 32

>>> shapefile.__version__
'1.2.10'

**Comment by karimbahgat** [comment]
Based on the error message this is not related to an empty shapefile. Rather, it is related to the code reaching the end of file (returning fewer than the needed 32 bytes) when reading the fields in the dbf file, which would make sense if there were no records after the header. But the error is because the lib believes there are more fields than there actually is. This could be because the calculation code is wrong, but this has not been a problem until now, and I have tested that reading empty shapefiles indeed works, even with zero fields. So more likely I suspect you might have had a file where the dbfheaderlength was misreported. Closing this for now, unless you have reason to believe this is a systematic error. 

(It could also be a rare issue with the dbf deletion field not being present if that is at all possible, but that would require diving into the dbf spec).
